### PR TITLE
cmd/cl-adm: Add log-level flag

### PR DESCRIPTION
**Pull request by kfirtoledo**
Add log-level flag to control the level of log messages.

### CODE REVIEW

File: tests/e2e/k8s/util/fabric.go [hunk 1]
```diff
@@ -38,6 +39,8 @@ type PeerConfig struct {
 	DataplaneType string
 	// Dataplanes is the number of dataplane instances.
 	Dataplanes uint16
+	// LogLevel is the log level.
+	LogLevel string
```
Comment (orozery) [inline tests/e2e/k8s/util/fabric.go:43]: I don't think we need this.
You can read the "DEBUG" "CICD" env variables at `Fabric::generateK8SYAML`, and pass it on directly to `platform.K8SConfig`.

File: pkg/bootstrap/platform/config.go [hunk 2]
```diff
@@ -37,11 +37,18 @@ type Config struct {
...
+	LogLevel string
 }
 
 const (
 	// DataplaneTypeEnvoy represents an envoy-type dataplane.
 	DataplaneTypeEnvoy = "envoy"
 	// DataplaneTypeGo represents a go-type dataplane.
 	DataplaneTypeGo = "go"
+	// InfoLogLevel represents an info log level.
+	InfoLogLevel = "info"
```
Comment (orozery) [inline pkg/bootstrap/platform/config.go:51]: I don't think we need these definitions. The logging level strings are pretty conventional.
Also, for envoy debugging TRACE level is much more helpful.
So I would use `trace` on `DEBUG=1`.

**Review by orozery — CHANGES_REQUESTED** [review]

File: tests/e2e/k8s/util/k8s_yaml.go [hunk 3]
```diff
@@ -107,18 +114,15 @@ func (f *Fabric) generateK8SYAML(p *peer, cfg *PeerConfig) (string, error) {
...
 	}
 
-	if os.Getenv("DEBUG") == "1" {
-		// enable debug logging
-		k8sYAML, err = enableControlplaneDebugLogs(k8sYAML)
-		if err != nil {
-			return "", fmt.Errorf("cannot enable controlplane debug logs: %w", err)
+	if (os.Getenv("DEBUG")) == "1" {
+		dpLogLevel := "trace" // More informative than the debug level.
+		if cfg.ExpectLargeDataplaneTraffic || os.Getenv("CICD") == "1" {
```
Comment (orozery) [inline tests/e2e/k8s/util/k8s_yaml.go:119]:
```suggestion
		if cfg.ExpectLargeDataplaneTraffic && os.Getenv("CICD") == "1" {
```

File: tests/e2e/k8s/util/k8s_yaml.go [hunk 4]
```diff
@@ -107,18 +114,15 @@ func (f *Fabric) generateK8SYAML(p *peer, cfg *PeerConfig) (string, error) {
...
 		}
 
-		if !cfg.ExpectLargeDataplaneTraffic || os.Getenv("CICD") != "1" {
-			k8sYAML, err = enableDataplaneDebugLogs(k8sYAML)
-			if err != nil {
-				return "", fmt.Errorf("cannot enable dataplane debug logs: %w", err)
-			}
+		k8sYAML, err = dataplaneDebugLogs(k8sYAML, dpLogLevel)
+		if err != nil {
+			return "", fmt.Errorf("cannot enable dataplane debug logs: %w", err)
```
Comment (orozery) [inline tests/e2e/k8s/util/k8s_yaml.go:125]:
```suggestion
			return "", fmt.Errorf("cannot set dataplane debug level: %w", err)
```

File: tests/e2e/k8s/util/k8s_yaml.go [hunk 5]
```diff
@@ -245,18 +249,10 @@ spec:
...
-func enableControlplaneDebugLogs(yaml string) (string, error) {
-	search := `controlplane
-          args: ["--log-level", "info"`
-	replace := `controlplane
-          args: ["--log-level", "debug"`
-	return replaceOnce(yaml, search, replace)
-}
-
-func enableDataplaneDebugLogs(yaml string) (string, error) {
+func dataplaneDebugLogs(yaml, logLevel string) (string, error) {
```
Comment (orozery) [inline tests/e2e/k8s/util/k8s_yaml.go:252]: Can you rename this function to `changeDataplaneDebugLevel`?

**Review by orozery — CHANGES_REQUESTED** [review]

**Comment by kfirtoledo** [comment]
@orozery, PTAL

**Review by orozery — APPROVED** [review]

**Pull request merged**